fmdtools/analyze/graph.py

```diff
@@ -1147,41 +1360,99 @@ def __init__(self, mdl, withstates=True, **kwargs):
         **kwargs : kwargs
             (placeholder for kwargs)
         """
-        self.g=self.nx_from_obj(mdl)
-        if withstates: self.set_nx_states(mdl)
+        self.g = self.nx_from_obj(mdl)
+        if withstates:
+            self.set_nx_states(mdl)
+
     def nx_from_obj(self, mdl):
-        g=mdl.graph.copy()
-        labels = {fname:f.get_typename() for fname, f in mdl.fxns.items()}
-        labels.update({fname:f.get_typename() for fname, f in mdl.flows.items()})
+        """
+        TODO Needs documentation
+        Parameters
+        ----------
+        mdl
+
+        Returns
+        -------
+
+        """
+        g = mdl.graph.copy()
+        labels = {fname: f.get_typename() for fname, f in mdl.fxns.items()}
+        labels.update({fname: f.get_typename() for fname, f in mdl.flows.items()})
         nx.set_node_attributes(g, labels, name='label')
         nx.set_edge_attributes(g, 'contains', name='label')
         return g
+
     def set_nx_states(self, mdl):
+        """
+        TODO : need documentation
+        Parameters
+        ----------
+        mdl
+
+        Returns
+        -------
+
+        """
         self.set_flow_nodestates(mdl)
         self.set_fxn_nodestates(mdl)
+
     def set_fxn_nodestates(self, mdl):
+        """
+        TODO : requires documentation
+        Parameters
```

Comment:
Attaches state attributes to Graph corresponding to the states of the model that belong to functions

mdl: Model
    Model to represent.

Reply:
Done